smithy-linters/src/main/java/software/amazon/smithy/linters/NoninclusiveTermsValidator.java

```diff
@@ -0,0 +1,224 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.linters;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.SourceLocation;
+import software.amazon.smithy.model.knowledge.TextIndex;
+import software.amazon.smithy.model.knowledge.TextInstance;
+import software.amazon.smithy.model.node.NodeMapper;
+import software.amazon.smithy.model.traits.Trait;
+import software.amazon.smithy.model.validation.AbstractValidator;
+import software.amazon.smithy.model.validation.Severity;
+import software.amazon.smithy.model.validation.ValidationEvent;
+import software.amazon.smithy.model.validation.ValidationUtils;
+import software.amazon.smithy.model.validation.ValidatorService;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.MapUtils;
+import software.amazon.smithy.utils.StringUtils;
+
+/**
+ * <p>Validates that all shape names, and values does not contain non-inclusive terms.
+ * *
+ * <p>See AbstractModelTextValidator for scan implementation details.
+ */
+public final class NoninclusiveTermsValidator extends AbstractValidator {
+    static final Map<String, List<String>> BUILT_IN_NONINCLUSIVE_TERMS = MapUtils.of(
+            "master", ListUtils.of("primary", "parent", "main"),
+            "slave", ListUtils.of("secondary", "replica", "clone", "child"),
+            "blacklist", ListUtils.of("denylist"),
+            "whitelist", ListUtils.of("allowlist")
+        );
+
+    public static final class Provider extends ValidatorService.Provider {
+        public Provider() {
+            super(NoninclusiveTermsValidator.class, node -> {
+                NodeMapper mapper = new NodeMapper();
+                return new NoninclusiveTermsValidator(
+                        mapper.deserialize(node, NoninclusiveTermsValidator.Config.class));
+            });
+        }
+    }
+
+    /**
+     * NoninclusiveTermsValidator configuration.
+     */
+    public static final class Config {
+        private Map<String, List<String>> noninclusiveTerms = MapUtils.of();
+        private boolean appendDefaults = false;
+
+        public Map<String, List<String>> getNonInclusiveTerms() {
+            return noninclusiveTerms;
+        }
+
+        public void setNoninclusiveTerms(Map<String, List<String>> terms) {
+            this.noninclusiveTerms = terms;
+        }
+
+        public boolean isAppendDefaults() {
+            return appendDefaults;
+        }
+
+        public void setAppendDefaults(boolean appendDefaults) {
+            this.appendDefaults = appendDefaults;
+        }
+    }
+
+    final Map<String, List<String>> termsMap;
```

Comment:
Should probably be `private` and initialized here, as that allows cleanup in the constructor as well.